src/Microsoft.AspNet.Mvc.Core/ActionResults/FileResult.cs

```diff
@@ -54,11 +55,12 @@ public override Task ExecuteResultAsync([NotNull] ActionContext context)
                 // detached and stored in a separate file. If the receiving MUA writes
                 // the entity to a file, the suggested filename should be used as a
                 // basis for the actual filename, where possible.
-                var headerValue = ContentDispositionUtil.GetHeaderValue(FileDownloadName);
-                context.HttpContext.Response.Headers.Set("Content-Disposition", headerValue);
+                var cd = new ContentDispositionHeaderValue("attachment");
+                cd.SetHttpFileName(FileDownloadName);
+                context.HttpContext.Response.Headers.Set("Content-Disposition", cd.ToString());
```

Comment:
response.ContentDisposition? No. response.GetTypedHeaders().ContentDisposition = cd would work, but it does the same thing as the above.